codyze-specification-languages/coko/coko-dsl/src/test/kotlin/de/fraunhofer/aisec/codyze/specificationLanguages/coko/dsl/CokoSarifBuilderTest.kt

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2024, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.fraunhofer.aisec.codyze.specificationLanguages.coko.dsl
+
+import de.fraunhofer.aisec.codyze.backends.cpg.CPGConfiguration
+import de.fraunhofer.aisec.codyze.backends.cpg.coko.CokoCpgBackend
+import de.fraunhofer.aisec.codyze.specificationLanguages.coko.core.CokoRule
+import de.fraunhofer.aisec.codyze.specificationLanguages.coko.core.Evaluator
+import de.fraunhofer.aisec.codyze.specificationLanguages.coko.core.dsl.Severity
+import de.fraunhofer.aisec.codyze.specificationLanguages.coko.core.toResultLevel
+import de.fraunhofer.aisec.codyze.specificationLanguages.coko.dsl.host.CokoExecutor
+import de.fraunhofer.aisec.cpg.passes.EdgeCachePass
+import de.fraunhofer.aisec.cpg.passes.UnreachableEOGPass
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import kotlin.io.path.Path
+import kotlin.reflect.KParameter
+import kotlin.reflect.KType
+import kotlin.reflect.KTypeParameter
+import kotlin.reflect.KVisibility
+
+class CokoSarifBuilderTest {
+
+    private val cpgConfiguration =
+        CPGConfiguration(
+            source = emptyList(),
+            useUnityBuild = false,
+            debugParser = false,
+            disableCleanup = false,
+            codeInNodes = true,
+            matchCommentsToNodes = false,
+            processAnnotations = false,
+            failOnError = false,
+            useParallelFrontends = false,
+            defaultPasses = true,
+            additionalLanguages = setOf(),
+            symbols = mapOf(),
+            includeBlocklist = listOf(),
+            includePaths = listOf(),
+            includeAllowlist = listOf(),
+            loadIncludes = false,
+            passes = listOf(UnreachableEOGPass::class, EdgeCachePass::class),
+        )
+
+    val cokoRulewithoutRuleAnnotation = object : CokoRule {
+        override val annotations: List<Annotation>
+            get() = emptyList()
+        override val isAbstract: Boolean
+            get() = TODO("Not yet implemented")
```

Comment:
I would rather directly throw a NotImplementedError instead of using a TODO() as it clutters the IDE with fake TODOs